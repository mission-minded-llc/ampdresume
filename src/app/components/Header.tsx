"use client";

import { Box } from "@mui/material";
import { useIsDesktop } from "@/hooks/useIsDesktop";
import { NavPrimary } from "./NavPrimary";
import { ThemeAppearanceToggle } from "./ThemeAppearanceToggle";

export const Header = () => {
  const isDesktop = useIsDesktop();

  return (
    <Box
      component="header"
      sx={(theme) => ({
        backgroundColor: "transparent",
        position: "sticky",
        zIndex: 9,
        top: 0,
        left: 0,
        width: "100vw",
        mb: 2,
        display: "flex",
        justifyContent: "space-between",
        alignItems: "center",
        paddingRight: isDesktop ? 2 : 0,
        [theme.breakpoints.down("sm")]: {
          mb: 0,
          flexDirection: "row-reverse",
        },
      })}
    >
      <NavPrimary />
      {isDesktop ? <ThemeAppearanceToggle /> : null}
    </Box>
  );
};
