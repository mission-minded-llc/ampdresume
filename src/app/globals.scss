@use "@/app/vars.scss" as *;

html,
body {
  max-width: 100vw;
  width: 100%;
  height: 100%;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

a {
  color: inherit;
}

ul {
  padding-inline-start: 0;
}

h3,
h4 {
  margin-bottom: 10px;
}

// Small tablet and mobile.
@media screen and (max-width: $breakpoint_mobile) {
  h2 {
    margin: 20px 0;
    padding: 20px 0;
  }
}
