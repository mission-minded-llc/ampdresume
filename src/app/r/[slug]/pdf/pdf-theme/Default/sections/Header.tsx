import { Typography } from "@mui/material";
import { User } from "@prisma/client";

export const Header = ({ user }: { user: User }) => (
  <>
    <Typography
      sx={{ padding: 1, fontSize: 13, textAlign: "center", borderBottom: `1px solid #ccc` }}
    >
      {user?.location ? `${user.location} | ` : null}
      {user?.displayEmail ? <a href={`mailto:${user.displayEmail}`}>{user.displayEmail}</a> : null}
    </Typography>
    <Typography sx={{ mt: 1, fontSize: 26, fontWeight: "bold", textAlign: "center" }}>
      {user.name}
    </Typography>
    <Typography
      sx={{ mb: 2, mt: 1, fontSize: 13, fontWeight: "bold", textAlign: "center", letterSpacing: 0 }}
    >
      {user.title}
    </Typography>
  </>
);
