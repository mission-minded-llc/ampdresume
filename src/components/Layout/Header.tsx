"use client";

import React from "react";

import { Box, Typography } from "@mui/material";
import Link from "next/link";
import { ThemeAppearanceToggle } from "./ThemeAppearanceToggle";

export const Header = () => {
  return (
    <Box
      component="header"
      sx={(theme) => ({
        backgroundColor: theme.palette.background.paper,
        position: "sticky",
        top: 0,
        left: 0,
        width: "100vw",
        boxShadow: "0 0 35px rgba(0, 0, 0, 0.1)",
        mb: "2em",
      })}
    >
      <Box
        sx={{
          width: "100%",
          display: "flex",
          justifyContent: "space-around",
          alignItems: "center",
        }}
      >
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
            gap: "0.35em",
          }}
        >
          <Typography component={Link} href="https://openresume.org">
            OpenResume
          </Typography>
        </Box>
        <Box component="nav">
          <Box
            sx={{
              display: "flex",
              gap: "1em",
            }}
          >
            <Typography component={Link} href="/account">
              Account
            </Typography>
          </Box>
        </Box>
        <ThemeAppearanceToggle />
      </Box>
    </Box>
  );
};
