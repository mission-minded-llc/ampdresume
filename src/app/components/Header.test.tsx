import "@testing-library/jest-dom";
import { useSession } from "next-auth/react";
import { render } from "@testing-library/react";
import { Header } from "./Header";
import { expect } from "@jest/globals";

jest.mock("next-auth/react");

describe("Header component", () => {
  it("matches snapshot", async () => {
    (useSession as jest.Mock).mockReturnValueOnce({
      data: {},
    });

    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
