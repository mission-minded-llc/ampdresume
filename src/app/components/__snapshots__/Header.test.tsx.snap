// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header component matches snapshot 1`] = `
<div>
  <header
    class="MuiBox-root css-1odyvdh"
  >
    <div
      class="MuiBox-root css-0"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1fgoo3o-MuiButtonBase-root-MuiIconButton-root"
        data-testid="NavPrimaryMenuIcon"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
      </button>
    </div>
  </header>
</div>
`;
